sql/planbuilder/dateparse/date_test.go

```diff
@@ -87,9 +87,9 @@ func TestConversionFailure(t *testing.T) {
 		expectedError string
 	}{
 		// with strict mode with NO_ZERO_IN_DATE,NO_ZERO_DATE enabled, these tests result NULL
-		{"no_year", "Jan 3", "%b %e", "0000-01-03", ""},
-		{"no_day", "Jan 2000", "%b %y", "2020-01-00", ""},
-		{"day_of_month_and_day_of_year", "Jan 3, 100 2000", "%b %e, %j %y", "2020-04-09", ""},
+		{"no_year", "Jan 3", "%b %e", time.Date(0, time.January, 3, 0, 0, 0, 0, time.UTC), ""},
+		{"no_day", "Jan 2000", "%b %y", time.Date(2019, time.December, 31, 0, 0, 0, 0, time.UTC), ""},
```

Comment:
This one confused me... shouldn't we return Jan 2000, instead of Dec 31st, 2019? 

When I tested this one on MySQL (version 9.0.1 at least, we may want to follow 8.4's behavior, but I didn't have that running currently), I got `NULL` for this query and the previous query.

Reply:
I believe that the test was a incorrect. It should be `%Y` instead of `%y` (the test right after is also wrong for the same reason). After that, I think it's some weirdness with the golang `time` package. There's no 0 day, so it just goes back to the previous year. 

Oddly, it seems like `time.Date(1999, time.December, 31...) == time.Date(2000, time.January, 0, ...)`